core/src/main/java/org/apache/iceberg/SchemaParser.java

```
@@ -183,13 +183,16 @@ private static Type typeFromJson(JsonNode json) {
       return Types.fromPrimitiveString(json.asText());
 
     } else if (json.isObject()) {
-      String type = json.get(TYPE).asText();
-      if (STRUCT.equals(type)) {
-        return structFromJson(json);
-      } else if (LIST.equals(type)) {
-        return listFromJson(json);
-      } else if (MAP.equals(type)) {
-        return mapFromJson(json);
+      JsonNode typeObj = json.get(TYPE);
+      if (typeObj != null) {
+        String type =  typeObj.asText();
+        if (STRUCT.equals(type)) {
+          return structFromJson(json);
+        } else if (LIST.equals(type)) {
+          return listFromJson(json);
+        } else if (MAP.equals(type)) {
+          return mapFromJson(json);
+        }
```

If the type object is null, then I think we need to throw an exception, right?